catalog/transformer/catalog-transformer-streaming-lib/src/test/java/org/codice/ddf/transformer/xml/streaming/lib/TestSaxEventToXmlElementConverter.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.transformer.xml.streaming.lib;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import javax.xml.stream.XMLStreamException;
+
+import org.junit.Test;
+import org.xml.sax.SAXException;
+
+public class TestSaxEventToXmlElementConverter {
+
+    /*
+     * Some of the "reconstructedExpectations" in this class are different than the input. This is acceptable and expected,
+     * because the output just has to be semantically the same as the input, it doesn't have to be literally the same.
+     * For example, sometimes the namespaces are declared at different scopes, in a different order, comments are removed, etc.
+     */
+
+    @Test
+    public void testSaxEventToXmlElementConverterRedeclaredNamespaceUri()
+            throws XMLStreamException, IOException, SAXException {
+        String doubleDeclaredNamespaceUriSnippet =
+                //@formatter:off
+                  "<x xmlns:n1='foobar' xmlns='foobar'>"
+                + "    <good a='1' b='2' />"
+                + "    <good a='1' n1:a='2' />"
+                + "</x>";
+                //@formatter:on
+
+        String reconstructedExpectation =
+                //@formatter:off
+                  "<x xmlns='foobar'>"
+                + "    <good a='1' b='2'></good>"
+                + "    <good a='1' xmlns:ns1='foobar' ns1:a='2'></good>"
+                + "</x>";
+                //@formatter:on
+        TestSaxParser parser = new TestSaxParser();
+        String reconstructedXml = parser.parseAndReconstruct(doubleDeclaredNamespaceUriSnippet);
+        reconstructedExpectation = reconstructedExpectation.replaceAll("'", "\"");
+        assertThat(reconstructedExpectation, is(reconstructedXml));
```

Comment:
The SAX Parser passes the attributes as an `Attributes` object, which seems to basically be an array of some kind of very simple class. As far as I can tell, the `Attributes` object always has the attributes in right-to-left order, and the converter always writes out the attributes by looping through them (0 ... n) so it should always respect the order.